### PR TITLE
Geojson Diff Multiple Check Support 

### PR DESCRIPTION
**Pull request by Bentleysb**
### Description:

This adds the ability to get the flag difference for multiple checks at once to the GeoJSONDiffSubCommand. Previously when this was attempted all checks were diffed together, which could result in the overriding of value and erroneous results. The LogDiffSubCommand already has this capability. 

In addition, this PR refactors the handling of the 2D hash maps that hold the flag data. Previously the maps were initialized and passed into void methods to be populated. This practice is not preferable, as it makes the data manipulation harder to track. These methods have been altered to instead return hash maps that are then merged together. 

### Potential Impact:

None

### Unit Test Approach:

One of the unit test resources was altered to include a flag that was identical to a flag that is already present in both the source and target test files, other than having a different check name. If the GeojsonDiff does not properly handle multiple checks, this flag will override the original flag and no changes will be detected. Two unit tests have been altered to check that this new cloned flag is detected as a different check, and shows up as an additional subtracted flag. 

### Test Results:

Ran the GeoJSON on a set of 51 checks with 92281 flags. Pre-enhancemnt the diff only reported 85371 total items, due to duplicate ids overriding each other. Post-enhancemnt the correct total is reported.

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/checks/commands/AtlasChecksGeoJSONDiffSubCommand.java [hunk 1]
```diff
@@ -40,46 +43,60 @@ public AtlasChecksGeoJSONDiffSubCommand()
     }
 
     @Override
-    protected void mapFeatures(final File file, final HashMap map)
+    protected HashMap<String, HashMap<String, JsonObject>> mapFeatures(final File file)
```
Comment (nachtm) [inline src/main/java/org/openstreetmap/atlas/checks/commands/AtlasChecksGeoJSONDiffSubCommand.java:46]: 🔥

File: src/main/java/org/openstreetmap/atlas/checks/commands/AtlasChecksGeoJSONDiffSubCommand.java [hunk 2]
```diff
@@ -40,46 +43,60 @@ public AtlasChecksGeoJSONDiffSubCommand()
...
         try (InputStream inputStream = file.isGzipped()
                 ? new GZIPInputStream(new FileInputStream(file.getFile())) : file.read())
         {
+            // Parse the json
             final JsonObject json = getGson()
                     .fromJson(new JsonReader(new InputStreamReader(inputStream)), JsonObject.class);
-            json.get(FEATURES).getAsJsonArray()
-                    .forEach(feature -> map.put(feature.getAsJsonObject().get(ID).getAsString(),
-                            feature.getAsJsonObject()));
+            //
```
Comment (nachtm) [inline src/main/java/org/openstreetmap/atlas/checks/commands/AtlasChecksGeoJSONDiffSubCommand.java:55]: nit: delete this?

File: src/main/java/org/openstreetmap/atlas/checks/commands/AtlasChecksGeoJSONDiffSubCommand.java [hunk 3]
```diff
@@ -40,46 +43,60 @@ public AtlasChecksGeoJSONDiffSubCommand()
...
+            json.get(FEATURES).getAsJsonArray().forEach(feature ->
+            {
+                final JsonObject jsonFeature = feature.getAsJsonObject();
+                // Get the check name
+                final Matcher nameMatch = NAME_PATTERN.matcher(
+                        jsonFeature.get(PROPERTIES).getAsJsonObject().get(NAME).getAsString());
+                nameMatch.find();
```
Comment (nachtm) [inline src/main/java/org/openstreetmap/atlas/checks/commands/AtlasChecksGeoJSONDiffSubCommand.java:62]: What does the regex do here?

**Review by nachtm — APPROVED** [review]
Big big big fan of the changes -- thanks for cleaning that up!

Quick nit and a question:

File: src/main/java/org/openstreetmap/atlas/checks/commands/AtlasChecksGeoJSONDiffSubCommand.java (hunk 3 above)
Comment (Bentleysb) [reply src/main/java/org/openstreetmap/atlas/checks/commands/AtlasChecksGeoJSONDiffSubCommand.java:62]: The name property contains the check name, but can also have an associated highway value. This grabs the check name, ignoring the possible highway tag. I will add a comment about this.

Comment (nachtm) [reply src/main/java/org/openstreetmap/atlas/checks/commands/AtlasChecksGeoJSONDiffSubCommand.java:62]: Makes sense -- thanks!

**Comment by Bentleysb** [comment]
Closing this PR in favor of one that will include fixes for breaking changes. Sorry for the timing @nachtm.

**Pull request closed without merging**